{{cookiecutter.dot}}/.biobuddies/justfile

```diff
@@ -0,0 +1,217 @@
+giha:
+    git describe --abbrev=40 --always --dirty --match=-
+
+# Backwards compatibility alias for giha
+gash: giha
+
+# print ORGanizatioN, a four letter acronym
+orgn:
+    #!/usr/bin/env bash
+    if [[ -n ${GITHUB_REPOSITORY_OWNER-} ]]; then
+        echo "$GITHUB_REPOSITORY_OWNER";
+    else
+        git remote get-url origin | sed -E 's,.+github.com/([^/]+).+,\1,';
+    fi
+
+# run Pre-Commit on modified files
+pc *args:
+    pre-commit run {{args}}
+
+# run Pre-Commit on All files
+pca *args:
+    pre-commit {{ \
+        if `just cona` == 'helicopyter' { 'try-repo .' } else { 'run' } \
+    }} --all-files {{args}}
+
+# run Pre-Commit on All files including Manual stage hooks
+pcam *args:
+    just pca --hook-stage manual {{args}}
+
+# run Pre-Commit on modified files including Manual stage hooks
+pcm *args:
+    pre-commit run --hook-stage manual {{args}}
+
+# git PUSH over https
+push *args:
+    git push https://${GITHUB_TOKEN}@github.com/$(just orgn)/$(just cona) {{args}}
+
+# Uv Pip Compile
+upc *args:
+    uv pip compile --all-extras --output-file requirements.txt --python-platform linux \
+        pyproject.toml {{ \
+            if path_exists('requirements.in') == 'true' { 'requirements.in' } else { '' } \
+        }} {{args}}
+
+# Update asdf .tool-versions
+update:
+    echo -e "glow $(asdf latest glow)\nuv $(asdf latest uv)\ntenv $(asdf latest tenv)" > .tool-versions
+
+# Uv venv and Pip Sync
+ups *args:
+    uv venv && uv pip sync {{args}} requirements.txt
+
+# Helicopyter Synth
+hs cona='all':
+    python -m helicopyter --format_with=${INSH_TF:-terraform} {{cona}}
+
+# Helicopyter synth and Terraform Apply
+hta cona envi *args:
+    #!/usr/bin/env bash
+    if [[ "{{envi}}" == "default" ]]; then
+        echo 'The default workspace behaves inconsistently.'
+        echo 'If you only have one environment, please name it `prod`.'
+        exit 1
+    fi
+    just hs {{cona}} \
+        && TF_VAR_giha=$(just giha) TF_VAR_tabr=$(just tabr) TF_WORKSPACE={{envi}} \
+            ${INSH_TF:-terraform} -chdir=deploys/{{cona}}/terraform apply {{args}}
+
+# Helper for Terraform Init and synth
+hti cona *args:
+    ${INSH_TF:-terraform} -chdir=deploys/{{cona}}/terraform init {{args}}
+    just hs {{cona}}
+
+# Helicopyter synth and Terraform Plan
+htp cona envi *args:
+    #!/usr/bin/env bash
+    if [[ "{{envi}}" == "default" ]]; then
+        echo 'The default workspace behaves inconsistently.'
+        echo 'If you only have one environment, please name it `prod`.'
+        exit 1
+    fi
+    just hs {{cona}} \
+        && TF_WORKSPACE={{envi}} ${INSH_TF:-terraform} -chdir=deploys/{{cona}}/terraform plan {{args}}
+
+# Universally Unique IDentifier
+uuid:
+    @echo {{uuid()}}
+
+# SUMMARIZE environment by displaying four letter acronyms
+summarize:
+    bash includes.sh summarize
+
+# Backwards compatibility alias for summarize
+ghas: summarize
+
+# TODO test if these special cases can be dropped
```

Comment:
Can you more clearly state what "these special cases" refers to?

Reply:
Using GitHub environment variables instead of `git describe`. I think I'll just drop the comment instead.